addons/web/models/models.py

```diff
@@ -157,6 +157,52 @@ def web_read(self, specification: Dict[str, Dict]) -> List[Dict]:
                     for values in values_list:
                         values[field_name] = [x2many_data.get(id_) or {'id': id_} for id_ in values[field_name]]
 
+            elif field.type in ('reference', 'many2one_reference'):
+                values_by_id = {
+                    vals['id']: vals
+                    for vals in values_list
+                }
+                for record in self:
+                    if not record[field_name]:
+                        continue
+
+                    if field.type == 'reference':
+                        co_record = record[field_name]
+                    else:  # field.type == 'many2one_reference'
+                        co_record = self.env[record[field.model_field]].browse(record[field_name])
+
+                    if 'context' in field_spec:
+                        co_record = co_record.with_context(**field_spec['context'])
+
+                    fields_to_read = dict(field_spec.get('fields'))
```

Comment:
`field_spec.get('fields')` returns `None` if the key `'fields'` is not in `field_spec`, and`dict(None)` will crash...  Maybe you could introduce an `if 'fields' in field_spec` like above:
```py
if 'fields' in field_spec:
    reference_read = co_record.web_read(field_spec['fields'])
    co_record_exists = bool(reference_read)
else:
    co_record_exists = bool(co_record.exists())
```